modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp

```diff
@@ -317,7 +317,7 @@ renderResponse.setTitle(headerTitle);
 
 					<aui:input label="title" name="title" />
 
-					<aui:input label="file-name" name="fileName" type='<%= dlEditFileEntryDisplayContext.isFileNameVisible() ? "text" : "hidden" %>' />
+					<aui:input label="file-name" name="fileName" required="<%= fileVersion!= null && Validator.isNotNull(fileVersion.getExtension()) %>" type='<%= dlEditFileEntryDisplayContext.isFileNameVisible() ? "text" : "hidden" %>' />
```

Comment:
@AliciaGarciaGarcia Can you surround `fileVersion != null` in parens and add a space between the variable and the operator? It should be `<%= (fileVersion != null) && ...`.

Reply:
In general, all subexpressions that involve binary operators should always be surrounded by parens, and binary operators need to be surrounded by spaces.

Reply:
Thanks for pointing at it !!